todo/src/reducer.js

```diff
@@ -0,0 +1,48 @@
+import { ADD_TODO, DELETE_TODO, TOGGLE_COMPLETE } from './actions';
+
+let defaultState = {
+    todos: [
+        {
+            name: 'init',
+            completed: false,
+            id: Date.now()
```

Comment:
Data.now() is an interesting idea for creating id's. You'll find that even huge systems use the date in their ids somewhere. It make them _iterable_ and _orderable_.